central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -381,15 +385,29 @@ func (m *managerImpl) getScanConfigWatcher(ctx context.Context, results *watcher
 	if sc == nil {
 		return nil, nil, false, errors.Errorf("ScanConfiguration not found for scan %s", results.Scan.GetScanName())
 	}
-	var w watcher.ScanConfigWatcher
-	var watcherIsRunning bool
-	concurrency.WithLock(&m.watchingScanConfigsLock, func() {
-		if w, watcherIsRunning = m.watchingScanConfigs[sc.GetId()]; !watcherIsRunning {
-			log.Debugf("Staring config watcher %s", sc.GetId())
+	w, watcherIsRunning := concurrency.WithLock2[watcher.ScanConfigWatcher, bool](&m.watchingScanConfigsLock, func() (watcher.ScanConfigWatcher, bool) {
+		w, watcherIsRunning := m.watchingScanConfigs[sc.GetId()]
+		return w, watcherIsRunning
```

Comment:
Would that be possible?
```suggestion
		return m.watchingScanConfigs[sc.GetId()]
```

Reply:
I just tested: 

```golang
func(key string, m map[string]int) (int, ok) {
    return m[key]
}
```

and it doesn't compile.

Reply:
Interesting... I would need to read more to understand how it selects whether to return 1 or 2 values.